src/main/java/org/broadinstitute/dsm/route/KitUploadRoute.java

```diff
@@ -479,29 +486,39 @@ private int getLastNonEmptyRowIndex(String[] rows) {
         return lastNonEmptyRowIndex;
     }
 
-    private boolean userExistsInRealm(DDPInstance ddpInstanceByRealm,
-                                      Map<String, String> participantDataByFieldName) {
-        String participantIdFromDoc = participantDataByFieldName.get(SHORT_ID);
-        String participantFirstNameFromDoc = participantDataByFieldName.get(FIRST_NAME);
-        String participantLastNameFromDoc = participantDataByFieldName.get(LAST_NAME);
+    private String userExistsInRealm(DDPInstance ddpInstanceByRealm,
+                                     Map<String, String> participantDataByFieldName) throws Exception{
+        String participantIdFromDoc = participantDataByFieldName.get(SHORT_ID).trim();
+        String participantFirstNameFromDoc = participantDataByFieldName.get(FIRST_NAME).trim().toLowerCase();
+        String participantLastNameFromDoc = participantDataByFieldName.get(LAST_NAME).trim().toLowerCase();
 
         Optional<ParticipantWrapper> maybeParticipant =
                 ParticipantWrapper.getParticipantByShortId(ddpInstanceByRealm, participantIdFromDoc);
 
         return isKitUploadNameMatchesToEsName(participantFirstNameFromDoc, participantLastNameFromDoc, maybeParticipant);
     }
 
-    boolean isKitUploadNameMatchesToEsName(String participantFirstNameFromDoc, String participantLastNameFromDoc,
-                              Optional<ParticipantWrapper> maybeParticipant) {
+    String isKitUploadNameMatchesToEsName(String participantFirstNameFromDoc, String participantLastNameFromDoc,
+                                          Optional<ParticipantWrapper> maybeParticipant) {
 
         Map<String, String> participantProfile = new HashMap<>();
         maybeParticipant.ifPresent(p -> {
             Map<String, String> participantProfileFromEs = (Map<String, String>) p.getData().get("profile");
-            participantProfile.put("firstName", participantProfileFromEs.get("firstName"));
-            participantProfile.put("lastName", participantProfileFromEs.get("lastName"));
+            participantProfile.put("firstName", participantProfileFromEs.get("firstName").trim().toLowerCase());
+            participantProfile.put("lastName", participantProfileFromEs.get("lastName").trim().toLowerCase());
         });
-        return participantFirstNameFromDoc.equals(participantProfile.get("firstName"))
-                && participantLastNameFromDoc.equals(participantProfile.get("lastName"));
+
+        String message = "";
+        if (!participantFirstNameFromDoc.equals(participantProfile.get("firstName"))) {
+            message += "First names in kit upload don't match the participant";
+            logger.error(message);
+        }
+
+        if (!participantLastNameFromDoc.equals(participantProfile.get("lastName"))) {
+            message += " Last names in kit Upload don't match the participant";
+            logger.error(message);
+        }
+        return message;
```

Comment:
Instead of returning message maybe throwing exceptions directly from if statements would make code more cleaner here and on line 433 as well, on line 433 would have only one `try/catch` block to handle exceptions.

Reply:
because that way the message we send to the frontend will show "RuntimeException".